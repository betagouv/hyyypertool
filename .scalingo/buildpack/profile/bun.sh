#!/usr/bin/env bash

export BUN_INSTALL="$HOME/.scalingo/bun"
export PATH="$BUN_INSTALL/bin:$PATH"
