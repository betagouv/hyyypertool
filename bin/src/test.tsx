//

import { Hono } from "hono";

//

export default new Hono().get("/", ({ html }) =>
  html(
    <html>
      <head></head>
      <body>From sdfssdfsdf ???</body>
    </html>,
  ),
);
