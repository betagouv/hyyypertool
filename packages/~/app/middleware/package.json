{
  "name": "@~/app.middleware",
  "version": "1.0.0",
  "private": true,
  "type": "module",
  "exports": {
    ".": {
      "default": "./src/index.ts"
    },
    "./session": {
      "default": "./src/session/index.ts"
    },
    "./*": {
      "default": "./src/*.ts"
    }
  },
  "dependencies": {
    "@~/app.core": "workspace:*",
    "@~/app.layout": "workspace:*",
    "@~/moncomptepro.database": "workspace:*",
    "hono": "4.3.9",
    "hono-sessions": "0.5.8",
    "openid-client": "5.6.5"
  },
  "devDependencies": {
    "@~/config.typescript": "workspace:*"
  }
}
