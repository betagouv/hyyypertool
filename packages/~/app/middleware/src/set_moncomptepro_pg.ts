//

import type { MonComptePro_PgDatabase } from "@~/moncomptepro.database";
import type { Env, MiddlewareHandler } from "hono";

//

export function set_moncomptepro_pg(
  client: MonComptePro_PgDatabase,
): MiddlewareHandler<MonComptePro_Pg_Context> {
  return async function set_moncomptepro_pg_middleware({ set }, next) {
    set("moncomptepro_pg", client);
    await next();
  };
}
//

export interface MonComptePro_Pg_Context extends Env {
  Variables: {
    moncomptepro_pg: MonComptePro_PgDatabase;
  };
}
