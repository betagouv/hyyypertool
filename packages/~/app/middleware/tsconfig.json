{
  "compilerOptions": {
    "jsxImportSource": "hono/jsx",
    "outDir": "./node_modules/.cache/tsc"
  },
  "extends": "@~/config.typescript/base.json",
  "references": [
    { "path": "../core/tsconfig.json" },
    { "path": "../sentry/tsconfig.json" },
    { "path": "../../infra/moncomptepro/database/tsconfig.json" }
  ]
}
