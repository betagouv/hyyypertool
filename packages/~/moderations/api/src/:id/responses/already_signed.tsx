//

import type { MonComptePro_Pg_Context } from "@~/app.middleware/moncomptepro_pg";
import { get_emails_by_organization_id } from "@~/users.repository/get_emails_by_organization_id";
import { useContext } from "hono/jsx";
import { useRequestContext } from "hono/jsx-renderer";
import { dedent } from "ts-dedent";
import { ModerationPage_Context } from "../context";

export const label = "Vous possédez déjà un compte MonComptePro";

export default async function template() {
  const { moderation } = useContext(ModerationPage_Context);

  const {
    var: { moncomptepro_pg },
  } = useRequestContext<MonComptePro_Pg_Context>();

  const members_email = await get_emails_by_organization_id(moncomptepro_pg, {
    organization_id: moderation.organizations.id,
  });

  return dedent`
    Bonjour,

    Votre demande pour rejoindre l'organisation « ${moderation.organizations.cached_libelle} » a été prise en compte sur https://app.moncomptepro.beta.gouv.fr.

    Vous possédez déjà un compte MonComptePro :

    - ${members_email.map(({ email }) => email).join("\n- ")}

    Merci de bien vouloir vous connecter avec le compte déjà existant.

    Je reste à votre disposition pour tout complément d'information.

    Excellente journée,
    L’équipe MonComptePro.
  `;
}
