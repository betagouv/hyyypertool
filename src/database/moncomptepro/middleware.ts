//

import env from ":common/env";
import { drizzle, type NodePgDatabase } from "drizzle-orm/node-postgres";
import type { Env, MiddlewareHandler } from "hono";
import { Client } from "pg";
import * as schema from "./drizzle/schema";

export function moncomptepro_pg_database({
  connectionString,
}: {
  connectionString: string;
}): MiddlewareHandler<moncomptepro_pg_Context> {
  return async function moncomptepro_pg_middleware({ req, set }, next) {
    const connection = new Client({ connectionString: connectionString });

    await connection.connect();

    const moncomptepro_pg = drizzle(connection, {
      schema,
      logger: env.DEPLOY_ENV === "preview",
    });

    set("moncomptepro_pg", moncomptepro_pg);

    await next();

    await connection.end();
  };
}

//

export interface moncomptepro_pg_Context extends Env {
  Variables: {
    moncomptepro_pg: NodePgDatabase<typeof schema>;
  };
}
