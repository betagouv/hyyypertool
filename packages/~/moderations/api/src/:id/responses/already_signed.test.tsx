//

import { set_moncomptepro_pg } from "@~/app.middleware/set_moncomptepro_pg";
import {
  create_adora_pony_user,
  create_pink_diamond_user,
  create_red_diamond_user,
  create_unicorn_organization,
} from "@~/moncomptepro.database/seed/unicorn";
import {
  add_user_to_organization,
  empty_database,
  migrate,
  pg,
} from "@~/moncomptepro.database/testing";
import { beforeAll, beforeEach, expect, test } from "bun:test";
import { Hono } from "hono";
import { jsxRenderer } from "hono/jsx-renderer";
import {
  type ContextType,
  type get_moderation_dto,
  type get_organization_member_dto,
} from "../context";
import already_signed from "./already_signed";

//

beforeAll(migrate);
beforeEach(empty_database);

test("returns all members", async () => {
  const unicorn_organization_id = await given_unicorn_organization();

  const app = new Hono<ContextType>()
    .use("*", jsxRenderer())
    .use("*", set_moncomptepro_pg(pg))
    .get(
      "/already_signed",
      ({ set }, next) => {
        set("domain", "unicorn.xyz");
        set("moderation", {
          organization: { cached_libelle: "ðŸ¦„", id: unicorn_organization_id },
          user: { family_name: "ðŸ§Ÿ" },
        } as get_moderation_dto);
        set("organization_member", {} as get_organization_member_dto);
        return next();
      },
      ({ render }) => {
        return render(<AlreadySigned />);
      },
    );

  const res = await app.fetch(
    new Request("http://localhost:3000/already_signed"),
  );
  expect(res.status).toBe(200);
  expect(await res.text()).toMatchSnapshot();
});

test("returns Diamond members", async () => {
  const unicorn_organization_id = await given_unicorn_organization();

  const app = new Hono<ContextType>()
    .use("*", jsxRenderer())
    .use("*", set_moncomptepro_pg(pg))
    .get(
      "/already_signed",
      ({ set }, next) => {
        set("domain", "unicorn.xyz");
        set("moderation", {
          organization: { cached_libelle: "ðŸ¦„", id: unicorn_organization_id },
          user: { family_name: "Diamond" },
        } as get_moderation_dto);
        set("organization_member", {} as get_organization_member_dto);
        return next();
      },
      ({ render }) => {
        return render(<AlreadySigned />);
      },
    );

  const res = await app.fetch(
    new Request("http://localhost:3000/already_signed"),
  );
  expect(res.status).toBe(200);
  expect(await res.text()).toMatchSnapshot();
});

async function given_unicorn_organization() {
  const unicorn_organization_id = await create_unicorn_organization(pg);
  const adora_pony_user_id = await create_adora_pony_user(pg);
  await add_user_to_organization({
    organization_id: unicorn_organization_id,
    user_id: adora_pony_user_id,
  });
  const pink_diamond_user_id = await create_pink_diamond_user(pg);
  await add_user_to_organization({
    organization_id: unicorn_organization_id,
    user_id: pink_diamond_user_id,
  });
  const red_diamond_user_id = await create_red_diamond_user(pg);
  await add_user_to_organization({
    organization_id: unicorn_organization_id,
    user_id: red_diamond_user_id,
  });
  return unicorn_organization_id;
}

async function AlreadySigned() {
  return <>{await already_signed()}</>;
}
