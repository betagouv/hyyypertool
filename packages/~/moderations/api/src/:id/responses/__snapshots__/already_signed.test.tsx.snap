// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`returns all members 1`] = `
"<!DOCTYPE html>Bonjour,

Votre demande pour rejoindre l&#39;organisation Â« ğŸ¦„ Â» a Ã©tÃ© prise en compte sur https://app.moncomptepro.beta.gouv.fr.

Vous possÃ©dez dÃ©jÃ  un compte MonComptePro :

- adora.pony@uni.corn
- pink.diamond@uni.corn
- red.diamond@uni.corn

Merci de bien vouloir vous connecter avec le compte dÃ©jÃ  existant.

Je reste Ã  votre disposition pour tout complÃ©ment d&#39;information.

Excellente journÃ©e,
Lâ€™Ã©quipe MonComptePro."
`;

exports[`returns Diamond members 1`] = `
"<!DOCTYPE html>Bonjour,

Votre demande pour rejoindre l&#39;organisation Â« ğŸ¦„ Â» a Ã©tÃ© prise en compte sur https://app.moncomptepro.beta.gouv.fr.

Vous possÃ©dez dÃ©jÃ  un compte MonComptePro :

- pink.diamond@uni.corn
- red.diamond@uni.corn

Merci de bien vouloir vous connecter avec le compte dÃ©jÃ  existant.

Je reste Ã  votre disposition pour tout complÃ©ment d&#39;information.

Excellente journÃ©e,
Lâ€™Ã©quipe MonComptePro."
`;
